### PR TITLE
show destination in statusBar & set udid in build

### PR DESCRIPTION
**Pull request by aelam**
* more detailed destination objects
  * anyMac 
  * myMac
  * any iPhone simulator
  * any iPhone device
  * listed simulator from `xcrun simctl`
  * listed devices
* add extra property isAvailableForBuild & isAvailableForRun, anyMac, any*** will not be available to run

* show destination in statusBar
   * it can select all destinations
   * When building an app, if the selected destination doesn't support this build, it will ask to select
* add pickDestination command
* add support for Mac build/Mac debug
* add support for watchos build only, debug doesn't work (because the paired watch+ iPhone simulator don't manage properly)

### CODE REVIEW

**Comment by aelam** [comment]
@hyzyla Do you have time to check this

**Comment by hyzyla** [comment]
> @hyzyla Do you have time to check this

Thanks for that PR. I've started reviewing and so far everything looks great. Great job! I need more time review and test everything locally and, I think, to end of week it will be merged and published to the marketplace

**Comment by aelam** [comment]
@hyzyla  Please note there are a lot of break changes. please lemme know where is not necessary 
for example, the panel for the simulator is disabled (because of a bug and unnecessity I think)

File: src/build/commands.ts [hunk 1]
```diff
@@ -201,30 +202,14 @@ function isXcbeautifyEnabled() {
...
-      default: {
-        const t: never = options.type;
-        throw new Error(`Unknown destination type: ${t}`);
-      }
-    }
+function getDestination(options: { platform: Platform; id: string | null }): string {
+  const platformName = getDestinationName(options.platform);
+  // ?? iPhone device can't be built with id
+  if (options.id != null && isSimulator(options.platform)) {
+    return `platform=${platformName},id=${options.id}`;
```
Comment (aelam) [inline src/build/commands.ts:209]: here should use identifier of real device not UDID

**Comment by hyzyla** [comment]
I apologize for the delayed response! I've just returned from vacation and have started working on your pull request. I decided to remove the "Simulators" and "Destinations" views and create a new "Destinations" view that includes all types of destinations. I will temporarily disable generic devices, macOS, and watchOS from the destinations because I want to refine the new "Destinations" view and your status bar for the next release. Of course, I will enable all other types of destinations from your pull request later. 

Here is quick preview how it looks like:
CleanShot 2024-07-20 at 17 59 01@2x

**Comment by hyzyla** [comment]
Thanks! 💜

**Comment by aelam** [comment]
Thank you!

**Pull request merged**